jdi-light/src/main/java/com/epam/jdi/light/elements/base/JDILocator.java

```diff
@@ -1,40 +1,40 @@
-package com.epam.jdi.light.common;
+package com.epam.jdi.light.elements.base;
 
 import com.epam.jdi.light.driver.WebDriverByUtils;
-import com.epam.jdi.light.elements.base.JDIBase;
+import com.epam.jdi.tools.Safe;
 import org.openqa.selenium.By;
 
 import java.util.List;
 
 import static com.epam.jdi.light.driver.WebDriverByUtils.*;
+import static com.epam.jdi.light.elements.base.JdiSettings.addTextToXPath;
 import static com.epam.jdi.light.settings.JDISettings.ELEMENT;
 import static com.epam.jdi.light.settings.WebSettings.printSmartLocators;
 import static com.epam.jdi.tools.LinqUtils.map;
 import static com.epam.jdi.tools.PrintUtils.print;
 import static java.lang.String.format;
-import static org.apache.commons.lang3.StringUtils.countMatches;
-import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.apache.commons.lang3.StringUtils.*;
 
 /**
  * Created by Roman Iovlev on 26.09.2019
  * Email: roman.iovlev.jdi@gmail.com; Skype: roman.iovlev
  */
 public class JDILocator {
-    public JDILocator() {}
-    public JDILocator(JDIBase element) { this.element = element; }
+    public JDILocator(JDIBase base) { this.element = new Safe<>(() -> base); }
     public JDILocator copy() {
-        JDILocator locator = new JDILocator();
+        JDIBase base = element.get();
+        JDILocator locator = new JDILocator(base);
         locator.byLocator = byLocator;
         locator.isRoot = isRoot;
-        locator.element = element;
+        locator.element = new Safe<>(() -> base);
         locator.frames = frames;
         return locator;
     }
 
     private By byLocator;
     private List<By> frames;
     public boolean isRoot = false;
-    private JDIBase element;
+    private Safe<JDIBase> element;
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6397461)